indexer/src/handler.rs

```diff
@@ -0,0 +1,187 @@
+                    self.dashmap.entry(tx.slot).or_insert(Vec::new()).push(tx);
+                },
+                Some(UpdateOneof::Slot(slot)) => {
+                    if let Some((_, transactions)) = self.dashmap.remove(&slot.slot) {
+                        for tx in transactions {
+                            task::spawn(self.clone().process_transaction(tx));
+                        }
+                    }
+                },
+                _ => {},
+            },
+            Err(error) => return Err(anyhow!("stream error: {:?}", error)),
+        };
+
+        Ok(())
+    }
+
+    async fn process_transaction(self, tx: SubscribeUpdateTransaction) -> Result<()> {
+        let message = tx
+            .transaction
+            .clone()
+            .context("SubsribeTransactionInfo not found")?
+            .transaction
+            .context("Transaction not found")?
+            .message
+            .context("Message not found")?;
+
+        let mut i = 0;
+        let keys = message.clone().account_keys;
+
+        for (idx, key) in message.clone().account_keys.iter().enumerate() {
+            let k = Pubkey::try_from(key.clone()).map_err(|_| anyhow!("failed to parse pubkey"))?;
+            if k == spl_token::ID {
+                i = idx;
+                break;
+            }
+        }
+
+        for ins in message.instructions.iter() {
+            let account_indices = ins.accounts.clone();
+            let program_idx: usize = ins.program_id_index.try_into()?;
+
+            if program_idx == i {
+                let data = ins.data.clone();
+                let data = data.as_slice();
+                let tkn_instruction = spl_token::instruction::TokenInstruction::unpack(data)?;
+                if let TokenInstruction::Transfer { amount } = tkn_instruction {
+                    if amount == 1 {
+                        let sig = tx
+                            .transaction
+                            .as_ref()
+                            .ok_or_else(|| anyhow!("failed to get transaction"))?
+                            .signature
+                            .clone();
+
+                        let source_account_index = account_indices[0];
+                        let source_bytes = &keys[source_account_index as usize];
+                        let source = Pubkey::try_from(source_bytes.clone())
+                            .map_err(|_| anyhow!("failed to parse pubkey"))?;
+                        let destination_account_index = account_indices[1];
+                        let destination_bytes = &keys[destination_account_index as usize];
+                        let destination = Pubkey::try_from(destination_bytes.clone())
+                            .map_err(|_| anyhow!("failed to parse pubkey"))?;
+
+                        let collection_mint =
+                            CollectionMint::find_by_ata(&self.db, source.to_string()).await?;
+
+                        if let Some(mint) = collection_mint {
```

Comment:
small thing but more readable to exit early when none so don't have the large code block nested in if condition.
```rust
if collection_mint.is_none() {
  return Ok(())
}
```